docs/sdks/js-ts/overview.mdx

```diff
@@ -60,139 +63,166 @@ In Kwil, accounts are identified by a public key. Kwil supports Secp256k1 Public
 
-Any action that executes a CUD operation must be signed and broadcasted to the network as a transaction.
+Any action that executes a CUD operation must be signed and broadcasted to the network through the `kwil.execute()` method.
 
-``` javascript
-import { Utils } from '@kwilteam/kwil-js'
+`.execute()` takes an object that matches the `ActionBody` interface. Action body has two required fields: `dbid` and `action`. You can also optionally add an `inputs` field if the action requires inputs, and a `description` field to customize the signature message.
+
+<Tabs
+    defaultValue="js"
+    values={[
+        { label: 'JavaScript', value: 'js', },
+        { label: 'Typescript', value: 'ts', },
+    ]}
+>
+<TabItem value="js">
+
+```javascript
+import { Utils } from "@kwilteam/kwil-js";
 
 // begin constructing the values for the action
 const input = new Utils.ActionInput()
-    .put("input_name_1", "input_value_1")
-    .put("input_name_2", "input_value_2")
-    .put("input_name_3", "input_value_3")
+  .put("input_name_1", "input_value_1")
+  .put("input_name_2", "input_value_2")
+  .put("input_name_3", "input_value_3");
 
-// retrieve database ID to locate action
-const dbid = kwil.getDBID("publicKey", "database_name")
+// construct action body
+const actionBody = {
+  dbid,
+  action: "your_action_name",
+  inputs: input,
+  description: "Click sign to execute the action!",
+};
 
-// construct and sign action transaction
-const tx = await kwil
-    .actionBuilder()
-    .dbid(dbid)
-    .name("your_action_name")
-    .concat(input)
-    .publicKey('public_key')
-    .signer(await provider.getSigner()) // can use wallet if NodeJS
-    .description("Click sign to execute the action!") // your custom signature message to appear in browser wallet
-    .buildTx()
-
-// broadcast transaction to kwil network
-const res = await kwil.broadcast(tx)
+// pass action body and signer to execute method
+const res = await kwil.execute(actionBody, kwilSigner);
 
 /*
     res.data = {
-        tx_hash: "some_hash",
+        tx_hash: "hash",
     }
 */
 ```
 
-#### Custom Signers
-
-The `.signer()` method also allows you to pass a custom signing function (must receive and return a `Uint8Array`) and the enumerator for the signature type used.
-
-Currently, Kwil supports three signature types:
+</TabItem>
+<TabItem value="ts">
 
-| Type  | Enumerator |
-|:----- |:------:|
-| Secp256k1  | 'secp256k1_ep' |
-| Ed25519    | 'ed25519' |
-| Ed25519 w/ NEAR Digest | 'ed25519_nr' |
-
-To sign with a ed25519 signature:
-
-```javascript
-import nacl from 'tweetnacl';
+```typescript
+import { Utils, Types } from "@kwilteam/kwil-js";
 
-const keys = nacl.sign.keyPair();
-const customSigner = (msg) => nacl.sign.detached(msg, keys.secretKey);
+// begin constructing the values for the action
+const input: Types.ActionInput = new Utils.ActionInput()
+  .put("input_name_1", "input_value_1")
+  .put("input_name_2", "input_value_2")
+  .put("input_name_3", "input_value_3");
```

Comment:
Why don't you just use an object?

```typescript
{
input_name_1: "input_value_1",
input_name_2: ...
}
```

Seems much easier than bringing in an extra utils package that has a builder

Reply:
You can.

The pass inputs, you can use the `ActionInput.put()`, `ActionInput.putFromObject()`, `ActionInput.putFromObjects()`, or just pass an array of objects in the `ActionBody` interface.

Should I just show the array of inputs?

Reply:
I feel like having `ActionInput.putFromObject()` is still weird.

Just do

```typescript
// construct action body
const actionBody: Types.ActionBody = {
  dbid: dbid,
  action: "your_action_name",
  inputs:{
     input_name_1: "input_value_1",
     input_name_2: ...
  },
  description: "Click sign to execute the action!",
};
```

Reply:
It removes the end user from having to import a utils package, and become aware of this `ActionInput` class